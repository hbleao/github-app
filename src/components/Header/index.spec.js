import React from 'react';
import { render, screen } from '@testing-library/react';
import user from '@testing-library/user-event';
import renderer from 'react-test-renderer';

import { Header } from '.';

describe('Header', () => {
  it('should be able to render', () => {
    expect(Header).toBeInstanceOf(Function);
  });

  it('should be able to see header with logo', () => {
    render(<Header />);

    const logoElement = document.querySelector('.header__logo');
    expect(logoElement).toBeTruthy();
  });

  it('should be able to receive an property onClick', () => {
    const onClick = jest.fn();
    render(<Header onClick={onClick} />);

    const linkElement = screen.getByText(/voltar/gi);
    user.click(linkElement);

    expect(onClick).toBeCalled();
  });

  it('should be able to press button five times', () => {
    const onClick = jest.fn();
    render(<Header onClick={onClick} />);

    const linkElement = screen.getByText(/voltar/gi);
    user.click(linkElement);
    user.click(linkElement);
    user.click(linkElement);
    user.click(linkElement);
    user.click(linkElement);

    expect(onClick).toBeCalledTimes(5);
  });

  it('should be able to press button five times', () => {
    render(<Header />);

    const linkElement = screen.queryByText(/voltar/gi);
    expect(linkElement).toBeNull();
  });

  it('should be able to render snapshot', () => {
    const onClick = jest.fn();
    const component = renderer.create(<Header onClick={onClick} />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
