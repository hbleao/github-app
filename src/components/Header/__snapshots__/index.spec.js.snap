// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should be able to render snapshot 1`] = `
<header
  className="header"
>
  <div
    className="header__logo"
  />
  <div
    className="header__link"
  >
    <div
      className="header__link-icon"
    />
    <button
      className="header__link-text"
      onClick={[MockFunction]}
    >
      Voltar
    </button>
  </div>
</header>
`;
